### PR TITLE
Dark theme

### LINKED ISSUE #51: Add a new theme
Add a new theme (may be dark theme)

### PR DESCRIPTION
**Pull request by IraSoro**
Closed #51 

I added a dark theme. To do this, I rewrote the code a little to make it easier to continue adding new themes.  It looks like this:
image

But I have a problem now. If the user has selected a dark theme and opens the application, then a highlight occurs, where the default theme (light) is first shown, then the dark one. I tried to fix this using `useLayoutEffect`, but it didn't work.

Also, I couldn’t adapt the `ionSelector` to the dark theme (I couldn’t change the background)

### CODE REVIEW

File: src/state/Context.tsx [hunk 1]
```diff
@@ -11,4 +11,15 @@ const cyclesInit: CyclesContextI = {
   updateCycles: (_newCycles) => {},
 };
 
+interface ThemeContextI {
```
Comment (imblowfish) [inline src/state/Context.tsx:14]: Usually interface name started with I, not ended

File: src/modals/Menu.tsx [hunk 2]
```diff
@@ -77,10 +80,49 @@ const LanguageSwitcher = () => {
   );
 };
 
+const ThemeSwitcher = () => {
+  const { t } = useTranslation();
+  const { theme, updateTheme } = useContext(ThemeContext);
+
+  const themesList = [];
+  for (const item of ["basic", "dark"]) {
```
Comment (imblowfish) [inline src/modals/Menu.tsx:88]: Call this theme dark(beta), because we have some problems with it for now

**Review by imblowfish — CHANGES_REQUESTED** [review]
I wrote some comments
About other problems, I think it should be ok to merge this theme as it now and add additional tasks to fix these bugs
Until then the dark theme may stay in beta

File: src/state/Context.tsx (hunk 1 above)
Comment (IraSoro) [reply src/state/Context.tsx:14]: Fixed

File: src/modals/Menu.tsx (hunk 2 above)
Comment (IraSoro) [reply src/modals/Menu.tsx:88]: Fixed

**Review by imblowfish — APPROVED** [review]

**Pull request merged**